app/models/workflow.rb

```diff
@@ -25,8 +25,18 @@ def external_signal?
     template&.signal_setting.present?
   end
 
+  def metadata
+    super.merge(:object_dependencies => object_dependencies)
+  end
+
   private
 
+  def object_dependencies
+    Hash.new { |h, k| h[k] = [] }.tap do |h|
+      tag_links.pluck(:app_name, :object_type).uniq.each { |k, v| h[k] << v }
+    end
```

Comment:
This is kind of difficult to digest. What do we think of this? Better? Worse?
```
Hash.new.tap do |dependencies|
  tag_links.pluck(:app_name, :object_type).uniq.each do |key, value|
    dependencies[key] ||= []
    dependencies[key] << value
  end
end
```

Reply:
@eclarizio 👍